### ISSUE TITLE
Regression from 2.13 when interpolating CMYK colours

### ISSUE THREAD

**Issue by amyspark** [mm2/Little-CMS#345]
Hi @mm2,

We've gotten [this bug](https://bugs.kde.org/show_bug.cgi?id=462356) reported against Krita 5.1.3, where we upgraded Little CMS to 2.14.

Between 2.13.1 and 2.14, something broke in the interpolation of CMYK colors, at least in 8-bit depth:

| 2.13.1 | 2.14 |
|-----|------|
| ![imagen](https://user-images.githubusercontent.com/13498015/204345237-1361bcdd-2f0d-44a4-8d64-acef9d884ac4.png)  | ![imagen](https://user-images.githubusercontent.com/13498015/204345188-7f252117-8c8d-4656-84f8-68d618b77605.png)  |

I'm not sure where in the commit log it happened, but disabling the fast float plugin has no effect. The offending profile is: https://bugs.kde.org/attachment.cgi?id=154115

**Comment by amyspark** [comment]
It only affects CMYK with 8-bit integer depth, 16-bit and float appear to be unaffected.

**Comment by mm2** [comment]
Checked other intents too, and BPC as well. All works fine to me.

**Comment by amyspark** [comment]
> the profile is used as input or output?

Input.

>    which intent?

Perceptual.

>    any special flags, black point compensation?

Only black point compensation and the fast float plugin enabled.

>    to which / from which profile are you doing the transform?

sRGB-elle-V2-srgbtrc (output, 8-bit integer depth).

I'll try to bisect now and see if I can trap the regression.

**Comment by mm2** [comment]
The plain CMM without plugin works fine in this configuration. I have to build a tificc with the plugin, maybe the issue is in the plugin. Tomorrow will try to do that.

**Comment by amyspark** [comment]
Git bisected between lcms2.13.1 and current HEAD (with a couple edits to build the fast float plugin under Meson-- btw, thanks for having it!)

Seems that one of the optimizations went awry:

```
❯ git bisect bad
d24a40c9cc685b7145e8082606d5d9384d08e472 is the first bad commit
commit d24a40c9cc685b7145e8082606d5d9384d08e472
Author: Marti Maria <marti.maria@littlecms.com>
Date:   Sun Feb 20 16:07:41 2022 +0100

    Fix LUT based optimization on 8 bits

    Most times, optimization was discarded where it could be used. Monotonicity is not needed here. Added a small demo.

 plugins/fast_float/src/fast_8_tethra.c |   6 +-
 plugins/fast_float/testbed/demo_cmyk.c | 151 +++++++++++++++++++++++++++++++++
 2 files changed, 152 insertions(+), 5 deletions(-)
 create mode 100644 plugins/fast_float/testbed/demo_cmyk.c
```

**Comment by mm2** [comment]
This makes sense. 
Seems a bug in the plug-in to me.
The bug was hidden in 2.13.1 because the profile was rejected for the plug-in and handled by the core engine. Now it is accepted, so the bug shows up. Will try to fix it ASAP.

**Comment by mm2** [comment]
This is a nasty bug on the plugin. Only affects 8 bits and CMYK. I will temporally disable CMYK on it 8 bits since the speed improvement is very small.

**Comment by mm2** [comment]
See 328c226af654adb321cedaa5cbe470eebd10e021

**Comment by mm2** [comment]
Fixed in 2.15

**Comment by amyspark** [comment]
@mm2 judging by the commit log, it seems there was no better solution found yet, right?
